### ISSUE TITLE
Sonarr reading seasons or TVID

### ISSUE THREAD

**Issue by septemberwasp** [jorenn92/Maintainerr#1248]
**Describe the bug**
Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')

**To Reproduce**
Creating a rule with "Sonarr - Is (part of) latest airing/aired season"

**Expected behavior**
I would assume it would pull that information and act on it accordingly from the set rules

**Screenshots**
attached

**Device (please complete the following information):**
 - OS: Docker
 - Version 2.1.0

**Additional context**
[maintainerr]  |  03/09/2024 13:51:39  [INFO] [RuleExecutorService] Starting Execution of all active rules
[maintainerr]  |  03/09/2024 13:51:39  [INFO] [RuleExecutorService] Executing rules for 'Aging Seasons'
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 268592
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 354765
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 328690
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 328690
[maintainerr]  |  03/09/2024 13:53:45  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
[maintainerr]  |  03/09/2024 13:55:00  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 289108
[maintainerr]  |  03/09/2024 13:55:00  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
[maintainerr]  |  03/09/2024 13:55:00  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 253573
[maintainerr]  |  03/09/2024 13:55:00  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
[maintainerr]  |  03/09/2024 13:55:00  [WARN] [SonarrApi] Could not retrieve show by tvdb ID 253573
[maintainerr]  |  03/09/2024 13:55:00  [WARN] [SonarrGetterService] Sonarr-Getter - Action failed : Cannot read properties of undefined (reading 'seasons')
![ruless](https://github.com/user-attachments/assets/e379ab56-1dc2-4163-bdc9-835f620651f2)

**Comment by ydkmlt84** [comment]
@septemberwasp are you still having this problem?

**Comment by septemberwasp** [comment]
Yup. Just checked, same result

**Comment by ydkmlt84** [comment]
We are talking about this right now on Discord but it is not a Maintainerr problem. Not technically anyways. The Plex metadata agent is pulling in the wrong TVDB/TMDB ids for some shows/movies and it is causing issues. Some people have had luck with unmatching and matching again in the Plex UI. You would have to look at all of your Plex media to try and see which shows it is complaining about.

TLDR --  Maintainerr pulls the TVDBid from Plex and tells Sonarr that that is the show it is looking for. That way we are talking about the EXACT same show. When Plex reports the incorrect TVDBid, Sonarr can't find that show and gives us this error.

**Comment by septemberwasp** [comment]
My Plex is mangled then (no surprise) as the list is quite long. 

The work around then is grab the TVdb ID from the maintainerr logs, locate the series in plex and un/re matching it?

**Comment by ydkmlt84** [comment]
Yes. What I ended up doing, which I don't recommend but it is one way. I removed the library from Plex and readded it. Fixed 99% of my TVDB/TMDB issues.

It may be easier to go one by one, unless you have hundreds.

**Comment by ydkmlt84** [comment]
I know it has been a while but did you have a chance to get this somewhat straightened out?

**Comment by septemberwasp** [comment]
I have not yet no. The workarounds I have no experimented with yet

**Comment by ydkmlt84** [comment]
> I have not yet no. The workarounds I have no experimented with yet

Ok. Just checking in :)
